marklogic-client-api/src/main/java/com/marklogic/client/datamovement/ZipSplitter.java

```diff
@@ -0,0 +1,363 @@
+import com.marklogic.client.impl.DocumentWriteOperationImpl;
+import com.marklogic.client.io.BytesHandle;
+import com.marklogic.client.io.Format;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Spliterator;
+import java.util.Spliterators;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.function.Predicate;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+
+/**
+ * The ZipSplitter class is used to split compressed files.
+ * The zip file could include XML, JSON, TXT or BINARY files.
+ */
+public class ZipSplitter implements Splitter<BytesHandle> {
+    private Map<String, Format> extensionFormats;
+    private Predicate<ZipEntry> entryFilter;
+    private Function<String, String> uriTransformer;
+    private int count = 0;
+
+    /**
+     * Returns the extensionFormats set to splitter. The extensionFormat is pre-defined in splitter.
+     * It includes "json", "txt" and "xml" extensions. If the file has no extension, it is treated as binary file.
+     * @return a map of extensionFormats
+     */
+    public Map<String, Format> getExtensionFormats() {
+        return this.extensionFormats;
+    }
+
+    /**
+     * Returns the entryFileter set to splitter.
+     * @return the entryFileter set to splitter
+     */
+    public Predicate<ZipEntry> getEntryFilter() {
+        return this.entryFilter;
+    }
+
+    /**
+     * Used to set entryFilter to splitter.
+     * @param entryFilter the filter that applied to each zipEntry
+     */
+    public void setEntryFilter(Predicate<ZipEntry> entryFilter) {
+        this.entryFilter = entryFilter;
+    }
+
+    /**
+     * Returns the uriTransformer set to splitter.
+     * @return the uriTransformer set to splitter
+     */
+    public Function<String, String> getUriTransformer() {
+        return this.uriTransformer;
+    }
+
+    /**
+     * Used to set uriTransformer to splitter
+     * @param uriTransformer the uriTransformer which applied on each document URI
+     */
+    public void setUriTransformer(Function<String, String> uriTransformer) {
+        this.uriTransformer = uriTransformer;
+    }
+
+    /**
+     * Create a new ZIP splitter.
+     */
+    public ZipSplitter() {
+        extensionFormats = new HashMap<>();
+        extensionFormats.put("", Format.UNKNOWN);
+        extensionFormats.put("json", Format.JSON);
+        extensionFormats.put("txt", Format.TEXT);
+        extensionFormats.put("xml", Format.XML);
+    }
+
+    @Override
+    public long getCount() {
+        return this.count;
+    }
+
+    /**
+     * Takes a input stream of a ZIP file and convert it to a stream of BytesHandle.
+     * The ZIP file could contain XML, JSON, TXT and BINARY files.
```

Comment:
Could the JavaDoc let the developer know that the input stream must be a ZipInputStream?

Reply:
Added below description.
"The input stream must be a ZipInputStream, otherwise it will throw an exception."